core/src/main/java/org/elasticsearch/common/bytes/PagedBytesReference.java

```diff
@@ -451,4 +358,30 @@ public int available() throws IOException {
         }
 
     }
+
+    @Override
+    public BytesRefIterator iterator() {
+        final BytesRef slice = new BytesRef();
+        return new BytesRefIterator() {
+            int position = 0;
+            @Override
+            public BytesRef next() throws IOException {
+                if (position == 0 && offset == 0 && length > 0) {
+                    // we are a slice and we are starting somewhere in the middle of a page
+                    int fragmentSize = Math.min(length, PAGE_SIZE - (offset % PAGE_SIZE));
```

Comment:
isn't offset always 0 here, based on the `if` condition? So this substraction is always 0?